crates/net/network/src/transactions/config.rs

```diff
@@ -28,6 +33,33 @@ impl Default for TransactionsManagerConfig {
             transaction_fetcher_config: TransactionFetcherConfig::default(),
             max_transactions_seen_by_peer_history: DEFAULT_MAX_COUNT_TRANSACTIONS_SEEN_BY_PEER,
             propagation_mode: TransactionPropagationMode::default(),
+            propogation_policy: TransactionPropagationPolicy::default(),
+        }
+    }
+}
+
+/// Determines which peers pending transactions are propagated to.
+#[derive(Debug, Clone, Default, PartialEq, Eq, Display)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum TransactionPropagationPolicy {
+    /// Propagate transactions to all peers.
+    #[default]
+    All,
+    /// Propagate transactions to only trusted peers.
+    ///
+    /// Can be used to keep the mempool private on networks like op mainnet
+    /// while still gossiping the transaction pool to trusted peers.
+    Trusted,
+}
```

Comment:
these make sense, but as @ferranbt we'd like to make this more flexible, so I suggest we do some changes here.
we rename this enum to `TransactionPropagationKind` so that we can have serde and fromstr support for cli.

then introduce a trait for `TransactionPropagationPolicy` that has the filter function.
then we can impl that trait for `TransactionPropagationKind`
and either add a Box dyn instance to transactionmanager or introduce a generic, I think both options would be okay, the dyn approach would be easier to integrate and I don't think dyn dispatch really matters here